### PR TITLE
PulsePoint adapter for AMP.

### PR DESCRIPTION
**Pull request by anand-venkatraman**
Please review PulsePoint (http://pulsepoint.com) adapter for AMP.

### CODE REVIEW

File: ads/_config.js [hunk 1]
```diff
@@ -180,6 +181,10 @@ export const adPreconnect = {
     'https://mads.at.atwola.com',
     'https://aka-cdn.adtechus.com',
   ],
+  pulsepoint: [
+    'https://tag.contextweb.com',
+    'https://ads.contextweb.com'
```
Comment (cramforce) [inline ads/_config.js]: No need to repeat origin used above in script URL.

File: ads/pulsepoint.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    tag(global, data);
+  }
+}
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+function tag(global, data) {
+  writeScript(global, 'https://tag.contextweb.com/getjs.aspx?action=VIEWAD&cwpid=' + data.pid + '&cwtagid=' + data.tagid + '&cwadformat=' + data.width + 'X' + data.height);
```
Comment (cramforce) [inline ads/pulsepoint.js]: Can you use loadScript instead?

File: ads/pulsepoint.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            width: data.width,
+            height: data.height,
+            slot: data.slot,
+            targeting: targeting[data.slot]
+          });
+        }
+    };
+    new PulsePointHeaderTag(hbConfig).init();
+  });
+}
```
Comment (cramforce) [inline ads/pulsepoint.js]: Add missing newline.

File: ads/pulsepoint.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+function tag(global, data) {
+  writeScript(global, 'https://tag.contextweb.com/getjs.aspx?action=VIEWAD&cwpid=' + data.pid + '&cwtagid=' + data.tagid + '&cwadformat=' + data.width + 'X' + data.height); 
+}
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+function headerBidding(global, data) {
+  loadScript(global, 'https://ads.contextweb.com/ht.js', () => {
```
Comment (cramforce) [inline ads/pulsepoint.js]: Do you want to do this once per page or once per ad?

File: ads/_config.js (hunk 1 above)
Comment (anand-venkatraman) [inline ads/_config.js]: Fixed

File: ads/pulsepoint.js (hunk 2 above)
Comment (anand-venkatraman) [inline ads/pulsepoint.js]: Added ```encodeURIComponent```. The script loaded does ```document.write```, which doesnt play nice with asynchronously loaded script.

File: ads/pulsepoint.js (hunk 3 above)
Comment (anand-venkatraman) [inline ads/pulsepoint.js]: Added missing newline.

File: ads/pulsepoint.js (hunk 4 above)
Comment (anand-venkatraman) [inline ads/pulsepoint.js]: Once per page would be good, but since each ad is loaded into its own ```iframe```, not sure how to achieve this.

**Comment by anand-venkatraman** [comment]
The Google CLA agreement has been signed.

File: ads/pulsepoint.js (hunk 4 above)
Comment (cramforce) [inline ads/pulsepoint.js]: `computeInMasterFrame` can run a function only once for all frames and make the result available.

**Comment by cramforce** [comment]
Please fix output of `gulp lint`

**Comment by anand-venkatraman** [comment]
Fixed ```lint``` errors.

File: ads/pulsepoint.js (hunk 4 above)
Comment (anand-venkatraman) [inline ads/pulsepoint.js]: I was wrong earlier, I wasnt able to run that code on master frame, so going to leave it as is.

**Comment by cramforce** [comment]
There are a few type errors that can be fixed by editing `ads.extern.js`. See https://travis-ci.org/ampproject/amphtml/builds/143452242

Otherwise this looks good.

**Comment by anand-venkatraman** [comment]
Fixed the type errors. Travis build passed.

**Comment by cramforce** [comment]
LGTM

**Comment by cramforce** [comment]
It looks like the CLA hasn't gone through review yet. Please ping me back in a day or 2

**Comment by anand-venkatraman** [comment]
We got a confirmation email today, that the CLA is completed. The CLA was signed as a corporation under the company name - 'PulsePoint Inc'.

**Pull request merged**